radicle-crdt/src/clock.rs

```diff
@@ -53,6 +62,36 @@ impl From<u64> for Lamport {
     }
 }
 
+impl From<Lamport> for u64 {
+    fn from(arg: Lamport) -> Self {
+        arg.get()
+    }
+}
+
+/// Error decoding an operation from an entry.
+#[derive(Error, Debug)]
+pub enum LamportError {
+    #[error("invalid Lamport Clock value")]
+    Invalid(),
+}
+
+impl FromStr for Lamport {
+    type Err = LamportError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Self::try_from(s)
+    }
+}
+
+impl TryFrom<&str> for Lamport {
+    type Error = LamportError;
+
+    fn try_from(s: &str) -> Result<Self, Self::Error> {
+        let v = s.parse::<u64>().map_err(|_| LamportError::Invalid())?;
+        Ok(v.into())
+    }
+}
```

Comment:
It's a bit redundant to offer `TryFrom` when you have `FromStr` already, we don't normally do that.

Reply:
I've definitely used that before. Gives some flexibility to do `Lamport::try_from` or `parse`.

Reply:
Indeed git::Oid was the inspiration.  Its not redundant in use, it makes try_from more complete and supports FromStr because its what people are used to.